build/stf-run-ci/tasks/create_catalog.yml

```diff
@@ -13,35 +13,60 @@
     sto_bundle_info: "{{ generate_bundle_sto.stdout_lines[-1] | from_json }}"
     sgo_bundle_info: "{{ generate_bundle_sgo.stdout_lines[-1] | from_json }}"
 
+- name: Generate default package names if not present
+  when: sto_bundle_info is defined and sto_bundle_info.package_name is not defined
+  ansible.builtin.set_fact:
+    sto_bundle_info: "{{ sto_bundle_info | combine({'package_name':'service-telemetry-operator.v%s'|format(sto_bundle_info.operator_bundle_version)}) }}"
+    sgo_bundle_info: "{{ sgo_bundle_info | combine({'package_name':'smart-gateway-operator.v%s'|format(sto_bundle_info.operator_bundle_version)}) }}"
+
 - name: Create info variables from provided pre-built bundles (deploy from bundles)
   when: __deploy_from_bundles_enabled | bool and not __local_build_enabled | bool
   block:
-    - name: Get STO operator bundle info
-      ansible.builtin.command: oc image info {{ __service_telemetry_bundle_image_path }}
-      register: sto_prebuilt_image_info
-
-    - name: Get SGO operator bundle info
-      ansible.builtin.command: oc image info {{ __smart_gateway_bundle_image_path }}
-      register: sgo_prebuilt_image_info
+    - name: Get STO operator bundle CSV file
+      ansible.builtin.command: oc image extract {{ __service_telemetry_bundle_image_path }} --file /manifests/*clusterserviceversion.yaml
+      args:
+        chdir: "{{ base_dir }}/working/service-telemetry-framework-index/"
+
+    - name: Get SGO operator bundle CSV file
+      ansible.builtin.command: oc image extract {{ __smart_gateway_bundle_image_path }}  --file /manifests/*clusterserviceversion.yaml
+      args:
+        chdir: "{{ base_dir }}/working/service-telemetry-framework-index/"
+
```

Comment:
You also need to update the filenames in the next two tasks to use this.

Reply:
Thanks for explaining! Reading Ansible docs along with your hints has led me to the slurp module, which might work even better here: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/slurp_module.html

We're ultimately shoving the file contents into a variable anyway, so if this works it would save us from the zuul vs not divergence and having to keep track of where the file is.

I'm giving it a try, if it doesn't work then I'll go with the way you suggested.

Reply:
I realize now you already pushed a fix so I'll leave it as is. Thank you.

Reply:
With the slurp module, you need to do something like:

```
- slurp:
      file: ...
  register: slurped_file
 
- set_fact:
      my_var: "{{ slurped_file.content | b64decode | from_yaml }}"
```

I can provide some playbook for testing that, if you want to play with it.